src/Traits/PermissionTrait.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace Caffeinated\Shinobi\Traits;
+
+trait PermissionTrait
+{
+    /**
+     * Users and Roles can have many permissions
+     *
+     * @return Illuminate\Database\Eloquent\Model
+     */
+    public function permissions()
+    {
+        return $this->belongsToMany('\Caffeinated\Shinobi\Models\Permission')->withTimestamps();
+    }
+
+    /**
+     * Get (cached) permission slugs assigned to the user or role.
+     * Internal method, should be implemented by Model implementing this trait
+     *
+     * @return array
+     */
+    protected function allPermissions()
```

Comment:
I would say this should be updated to `getCachedPermissions`

Reply:
Sorry, the doc comment is wrong. `allPermissions` does actually do nothing with regards to caching. It is used by `getPermissions` which does caching, but allPermissions always retrieves the permissions from the db. So following Laravel terms maybe `getFreshPermissions` would be better?

For background info, I included the placeholder method in the trait to prevent php errors if one would use the PermissionTrait in their own Model. Unfortunately after this you'd probably still get errors anyway as a `shinobiTag` definition is not part of the trait, but afaik you cant overload any class variables using traits. Thinking about it, we could fix that by just including a `getShinobiTag` method instead, maybe that would be better?